.github/workflows/checks.yml

```diff
@@ -5,17 +5,31 @@ on:
     branches:
       - master
   pull_request:
+  schedule:
+    # At 16:00 on Wednesday
+    - cron: "0 16 * * 3"
 
 jobs:
   run-tests:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+
+      - name: Get scarb version
+        id: extractScarbVersion
+        run: |
+          curl -s https://api.github.com/repos/foundry-rs/starknet-foundry/releases/latest | grep tarball_url | cut -d '"' -f 4 | xargs -n 1 curl -L -o latest.tar.gz
+          mkdir latest
+          tar -xz --strip-components=1 -C latest -f latest.tar.gz
+          version=$(grep scarb latest/.tool-versions | awk '{print $2}')
```

Comment:
that is a very fine drutowanko
but I guess we could get rid of downloading tar making it a bit faster (although probably not less hacky)

```
snfoundry_version=$(curl -s https://api.github.com/repos/foundry-rs/starknet-foundry/releases/latest | grep tarball_url | awk -F '/' '{print $8}' | tr -d '",')

scarb_version=$(curl -s https://raw.githubusercontent.com/foundry-rs/starknet-foundry/$snfoundry_version/.tool-versions | awk '{print $2}')
```

or in one not so pretty line

```
scarb_version=$(curl -s https://raw.githubusercontent.com/foundry-rs/starknet-foundry/$(curl -s https://api.github.com/repos/foundry-rs/starknet-foundry/releases/latest | grep tarball_url | awk -F '/' '{print $8}' | tr -d '",')/.tool-versions | awk '{print $2}')
```

=====

Maybe a good workaround for this would be to modify snfoundry gh action to output not only its version, but scarb version based on this file as well? This is just w thought for the future, if we ever wanted to avoid such hacks

Reply:
Sure, just to clarify - I did not mean snfoundry should do it, but rather the action itself. It already has snfoundry code checked out, so all we would need to do is 1. parse the file; 2. Modify action output to include an information like "works_with_scarb_version=x.y.z"; 3. We then could use action output in another action to set up specific scarb version. 

This is just an idea though, totally up to you :+1: